boards/seeed/xiao_mg24/board.cmake

```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+
+board_runner_args(openocd)
+include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
+
+board_runner_args(jlink "--device=EFR32MG24BxxxF1536")
+include(${ZEPHYR_BASE}/boards/common/jlink.board.cmake)
```

Comment:
I suggest to submit [the OpenOCD patch][1] to [the Zephyr fork of OpenOCD][2].

[1]: https://github.com/facchinm/OpenOCD/commit/21fa2de702699ff647a5fc118718884b5d9f28fd
[2]: https://github.com/zephyrproject-rtos/openocd

Meanwhile, since the support for OpenOCD does not (yet) work out of the box, maybe we should place support for `jlink` before support for `openocd`?

Optionally, you can also add support for commander:
```
board_runner_args(silabs_commander "--device=EFR32MG24B220F1536")
include(${ZEPHYR_BASE}/boards/common/silabs_commander.board.cmake)
```
Commander is less powerful than Jlink because it can't debug the board. However, it tends to provide a better user experience with Silabs boards.

Reply:
What I said is wrong. This board does not include JLink. So, neither commander, neither Jlink works. Probably you want to remove the line about jlink.

My comment about adding support for Series-2 to Zephyr fork of OpenOCD is still valid.

Reply:
Created https://github.com/zephyrproject-rtos/openocd/pull/70 to bring in this support. Built locally (after some unrelated compile error fixes to other drivers), installed, and used to flash the XIAO MG24 successfully.

Reply:
Regarding removing jlink: The XIAO MG24 *does* expose SWD pads on the bottom side, so I had felt leaving this in, properly configured, would allow folks to use a jlink against this target if they attached one. I'm happy to remove it if that's preferred though.